lib/cucumber/multiline_argument/data_table.rb

```diff
@@ -246,56 +240,41 @@ def match(pattern)
       # A StepDefinition receiving this table can then map the columns
       # with both Regexp and String:
       #
-      #   table.map_headers!(/phone( number)?/i => :phone, 'Address' => :address)
+      #   table.map_headers(/phone( number)?/i => :phone, 'Address' => :address)
       #   table.hashes
       #   # => [{:phone => '123456', :address => 'xyz'}, {:phone => '345678', :address => 'abc'}]
       #
       # You may also pass in a block if you wish to convert all of the headers:
       #
-      #   table.map_headers! { |header| header.downcase }
+      #   table.map_headers { |header| header.downcase }
       #   table.hashes.keys
       #   # => ['phone number', 'address']
       #
       # When a block is passed in along with a hash then the mappings in the hash take precendence:
       #
-      #   table.map_headers!('Address' => 'ADDRESS') { |header| header.downcase }
+      #   table.map_headers('Address' => 'ADDRESS') { |header| header.downcase }
       #   table.hashes.keys
       #   # => ['phone number', 'ADDRESS']
       #
-      def map_headers!(mappings = {}, &block)
-        # TODO: Remove this method for 2.0
-        clear_cache!
-        @header_mappings = mappings
-        @header_conversion_proc = block
-      end
-
-      # Returns a new Table where the headers are redefined. See #map_headers!
       def map_headers(mappings = {}, &block)
         self.class.new(Core::Test::DataTable.new(raw), @conversion_procs.dup, mappings, block)
       end
 
+      # Returns a new Table with an additional column mapping.
+      #
       # Change how #hashes converts column values. The +column_name+ argument identifies the column
       # and +conversion_proc+ performs the conversion for each cell in that column. If +strict+ is
       # true, an error will be raised if the column named +column_name+ is not found. If +strict+
       # is false, no error will be raised. Example:
       #
       #   Given /^an expense report for (.*) with the following posts:$/ do |table|
-      #     posts_table.map_column!('amount') { |a| a.to_i }
+      #     posts_table = posts_table.map_column('amount') { |a| a.to_i }
       #     posts_table.hashes.each do |post|
       #       # post['amount'] is a Fixnum, rather than a String
       #     end
       #   end
       #
       # rubocop:disable Style/OptionalBooleanParameter # the optional boolean parameter is kept for retrocompatibility
-      def map_column!(column_name, strict = true, &conversion_proc)
-        # TODO: Remove this method for 2.0
-        @conversion_procs[column_name.to_s] = { strict: strict, proc: conversion_proc }
-        self
-      end
-      # rubocop:enable Style/OptionalBooleanParameter
-
-      # Returns a new Table with an additional column mapping. See #map_column!
-      # rubocop:disable Style/OptionalBooleanParameter # the optional boolean parameter is kept for retrocompatibility
       def map_column(column_name, strict = true, &conversion_proc)
```

Comment:
Given this rubocop offense is then sanitized down the stack, this feels like we should fix this here too (make it a keyword arg)

I'm aware this might technically make this a breaking change, so can defer if you would prefer

Reply:
Well, 8.0.0 will be full of breaking changes.
But maybe in its own PR?

Reply:
Oki, wanna jot that down or make a ticket for it? If so I can approve this PR.

Reply:
https://github.com/cucumber/cucumber-ruby/issues/1592